### PR TITLE
[Task-7] As a user, I can view sign in screen & [Task-9] As a signed in user, I can view sign out option

### PR DESCRIPTION
**Pull request by mcneto**
Task - [[Task-7] As a user, I can view sign in screen](https://github.com/mcneto/ruby-google-search-ic/issues/4)
Task - [[Task-9] As a signed in user, I can view sign out option](https://github.com/mcneto/ruby-google-search-ic/issues/10)

## What happened

- Implement bootstrap to sign in screen

## Insight

- Add sign in option to navigation bar
- Add sign out option to navigation bar
- Add style to all form and alert
- Forgot password is not functioning currently

## Proof Of Work
**Sign In / Sign Out**
ezgif com-gif-maker (2)

**Error Message**
Screen Shot 2564-07-30 at 19 51 55

### CODE REVIEW

**Review by rafayet-monon — COMMENTED** [review]
The same comments will be applied for this as #38 

- https://github.com/mcneto/ruby-google-search-ic/pull/38#discussion_r660318185
- https://github.com/mcneto/ruby-google-search-ic/pull/38#discussion_r660318890

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
 
-  <div class="field">
-    <%= f.label :password %><br/>
-    <%= f.password_field :password, autocomplete: 'current-password' %>
-  </div>
+  <div class="card-body">
+    <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+      <div class="form-group">
+        <%= f.label :email %><br/>
+        <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "form-control", placeholder: "Email" %>
```
Comment (junan) [inline app/views/devise/sessions/new.html.erb:10]: How about to add HTML input validation?
```suggestion
        <%= f.email_field :email, required: true, autofocus: true, autocomplete: "email", class: "form-control", placeholder: "Email" %>
```

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
+      </div>
+
+      <div class="form-group">
+        <%= f.label :password %><br/>
+        <%= f.password_field :password, autocomplete: "current-password", class: "form-control", placeholder: "Password" %>
```
Comment (junan) [inline app/views/devise/sessions/new.html.erb:15]: Same as above comment

File: spec/systems/sign_in_spec.rb [hunk 3]
```diff
@@ -3,6 +3,25 @@
...
+  context 'given the navigation bar' do
+    it 'displays the sign-in link' do
+      visit root_path
+      expect(find('nav')).to have_link('Sign In')
+    end
+  end
+
+  context 'given the sign-in page is visited' do
+    it 'displays the email field' do
+      visit new_user_registration_path
```
Comment (junan) [inline spec/systems/sign_in_spec.rb:15]: This file is for sign-in test, if we need to test registration, it should be on `registration_spec.rb`

File: spec/systems/sign_in_spec.rb [hunk 4]
```diff
@@ -3,6 +3,25 @@
 require 'rails_helper'
 
 describe 'Sign in', type: :system do
+  context 'given the navigation bar' do
+    it 'displays the sign-in link' do
+      visit root_path
```
Comment (junan) [inline spec/systems/sign_in_spec.rb:8]: As it is for sign-in test, lets use the direct sign in path here

File: spec/systems/sign_in_spec.rb [hunk 5]
```diff
@@ -3,6 +3,25 @@
...
+  end
+
+  context 'given the sign-in page is visited' do
+    it 'displays the email field' do
+      visit new_user_registration_path
+      expect(page).to have_field('Email')
+    end
+
+    it 'displays the password field' do
+      visit new_user_registration_path
```
Comment (junan) [inline spec/systems/sign_in_spec.rb:20]: Same as bove comment

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (mcneto) [reply app/views/devise/sessions/new.html.erb]: Added in ba5f1f7

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (mcneto) [reply app/views/devise/sessions/new.html.erb]: Added in ba5f1f7

File: spec/systems/sign_in_spec.rb (hunk 3 above)
Comment (mcneto) [reply spec/systems/sign_in_spec.rb]: Fixed in 0f4ad12

File: spec/systems/sign_in_spec.rb (hunk 4 above)
Comment (mcneto) [reply spec/systems/sign_in_spec.rb]: Fixed in 0f4ad12

File: spec/systems/sign_in_spec.rb (hunk 5 above)
Comment (mcneto) [reply spec/systems/sign_in_spec.rb]: Fixed in 0f4ad12

**Review by junan — APPROVED** [review]

File: spec/systems/sign_in_spec.rb [hunk 6]
```diff
@@ -3,6 +3,25 @@
 require 'rails_helper'
 
 describe 'Sign in', type: :system do
+  context 'given the navigation bar' do
+    it 'displays the sign-in link' do
```
[the remark this answers is not in the document]
Comment (mcneto) [reply spec/systems/sign_in_spec.rb:7]: Fixed in 734014c

**Review by rafayet-monon — APPROVED** [review]

**Pull request merged**